examples/drawing/plot_tsp.py

```diff
@@ -0,0 +1,53 @@
+"""
+==========================
+Traveling Salesman Problem
+==========================
+
+This is an example of a drawing solution of the traveling salesman problem
+
+The function is used to produce the solution is christofides,
+where given a set of nodes, it calculates the route of the nodes
+that the traveler has to follow in order to minimize the total cost.
+"""
+
+import matplotlib.pyplot as plt
+import networkx as nx
+import networkx.algorithms.approximation as nx_app
+import math
+
+G = nx.random_geometric_graph(20, radius=0.4, seed=3)
+pos = nx.get_node_attributes(G, "pos")
+
+# Depot should be at (0,0)
+pos[0][0], pos[0][1] = 0.5, 0.5
+
+H = G.copy()
+
+
+# Calculating the distances between the nodes as edge's weight.
+for i in range(len(pos)):
+    for j in range(i + 1, len(pos)):
+        dist = math.hypot(pos[i][0] - pos[j][0], pos[i][1] - pos[j][1])
+        dist = dist
+        G.add_edge(i, j, weight=dist)
```

Comment:
Alternatively you could explicitly create the complete graph (required by Christofides) explicitly and iterate through the edges there, e.g.

```python
G = nx.complete_graph(20)
for u, v in G.edges():
    G[u][v]['weight'] = math.hypot(pos[v][0] - pos[u][0], pos[v][1] - pos[u][1])
```